src/libstore/build/local-derivation-goal.cc

```diff
@@ -182,7 +184,15 @@ inline bool DerivationGoal::needsHashRewrite()
 }
 
 
-void DerivationGoal::killChild()
+LocalStore & LocalDerivationGoal::getLocalStore()
+{
+    auto p = dynamic_cast<LocalStore *>(&worker.store);
```

Comment:
Rather than this specialized method, maybe we could have a method `Store & DerivationGoal::getStore()` that would be overriden as `LocalStore & DerivationGoal::getStore()`

Reply:
Errr I'm bit a confused, you wan to replace all the `worker.store` with `getStore()`? This is method right now is just a regular one on the subclass, and not an override of anything on the base class.